allennlp/data/dataset_readers/dataset_utils/span_utils.py

```diff
@@ -136,6 +136,80 @@ def bio_tags_to_spans(tag_sequence: List[str],
     return list(spans)
 
 
+def iob1_tags_to_spans(tag_sequence: List[str],
+                       classes_to_ignore: List[str] = None) -> List[TypedStringSpan]:
+    """
+    Given a sequence corresponding to IOB1 tags, extracts spans.
+    Spans are inclusive and can be of zero length, representing a single word span.
+    Ill-formed spans are also included (i.e., those where "B-LABEL" is not preceded
+    by "I-LABEL" or "B-LABEL").
+
+    Parameters
+    ----------
+    tag_sequence : List[str], required.
+        The integer class labels for a sequence.
+    classes_to_ignore : List[str], optional (default = None).
+        A list of string class labels `excluding` the bio tag
+        which should be ignored when extracting spans.
+
+    Returns
+    -------
+    spans : List[TypedStringSpan]
+        The typed, extracted spans from the sequence, in the format (label, (span_start, span_end)).
+        Note that the label `does not` contain any BIO tag prefixes.
+    """
+    classes_to_ignore = classes_to_ignore or []
+    spans: Set[Tuple[str, Tuple[int, int]]] = set()
+    span_start = 0
+    span_end = 0
+    active_conll_tag = None
+    for index, string_tag in enumerate(tag_sequence):
+        prev_string_tag = tag_sequence[index-1] if index - 1 > -1 else None
```

Comment:
Instead of having these `if ... else` guards, just pull these definitions out of the loop, set them to `None` and update them at the end of the loop